### PR TITLE
[Merged by Bors] - Remove unnecessary stuff and get egl from a content snap

### LINKED ISSUE #8: Too big
Probably because of the inclusion of swaybg this is bigger than mir-kiosk:
```
$ ls -lh /var/lib/snapd/snaps/ubuntu-core-kiosk_*.snap /var/lib/snapd/snaps/mir-kiosk_*.snap
-rw------- 1 root root 47M Mar 22 11:33 /var/lib/snapd/snaps/mir-kiosk_6592.snap
-rw------- 1 root root 47M Mar 25 12:27 /var/lib/snapd/snaps/mir-kiosk_6615.snap
-rw------- 1 root root 53M May 17 16:53 /var/lib/snapd/snaps/ubuntu-core-kiosk_25.snap
-rw------- 1 root root 53M May 18 08:29 /var/lib/snapd/snaps/ubuntu-core-kiosk_29.snap
```

### PR DESCRIPTION
**Pull request by AlanGriffiths**
```
$ snap install --channel edge/pr9 ubuntu-core-kiosk
```
The corresponding mesa-core20 content snap is unlisted, but is automatically installed from the snap store. The source for the content snap is at:

    https://github.com/MirServer/mesa-core20

Fixes: #8
```
$ ls -lh *.snap
-rw-r--r-- 1 alan alan 6.4M May 21 12:39 ubuntu-core-kiosk_18-mir2.3.3+dev321_amd64.snap
```

### CODE REVIEW

**Comment by AlanGriffiths** [comment]
@wmww Landing this will make an easier workflow for @RAOF as it will automate a rebuild of the content consumer for all architectures with fixes that land on Mir master. (And he needs to land some.)

So, review with a "land and fixup later" preference.

File: src/ubuntu_core_kiosk_main.cpp [hunk 1]
```diff
@@ -17,63 +17,38 @@
...
-
     MirRunner runner{argc, argv};
 
     DisplayConfiguration display_config{runner};
     WaylandExtensions wayland_extensions;
 
-    for (auto const& extension : {
-        WaylandExtensions::zwlr_layer_shell_v1, // For swaybg
-        "zwp_pointer_constraints_v1",           // Useful for games
-        "zwp_relative_pointer_manager_v1"})     // Useful for games
```
Comment (wmww) [inline src/ubuntu_core_kiosk_main.cpp:54]: I assume we just want to drop layer shell, and not the pointer ones?

**Review by wmww — APPROVED** [review]
Sorry I missed this earlier
bors r+

**Comment by AlanGriffiths** [comment]
On Fri, 21 May 2021 20:48:57 -0700, William Wold wrote:
> @wmww approved this pull request.
>
> Sorry I missed this earlier
> bors r+
>
>>      MirRunner runner{argc, argv};
>  
>      DisplayConfiguration display_config{runner};
>      WaylandExtensions wayland_extensions;
>  
> -    for (auto const& extension : {
> -        WaylandExtensions::zwlr_layer_shell_v1, // For swaybg
> -        "zwp_pointer_constraints_v1",           // Useful for games
> -        "zwp_relative_pointer_manager_v1"})     // Useful for games
>
> I assume we just want to drop layer shell, and not the pointer ones?
>

Actually, we want to drop them. They cause problems for electron clients. And enabling us easy with the new config tools.

-- 
Sent using Dekko from my Ubuntu device

**Pull request closed without merging**